lib/internal/http2/core.js

```diff
@@ -1425,11 +1425,14 @@ function connectionListener(socket) {
     socket.on('timeout', socketOnTimeout);
   }
 
-  // TLS ALPN fallback to HTTP/1.1
-  if (options.allowHTTP1 === true &&
-    (socket.alpnProtocol === false ||
-    socket.alpnProtocol === 'http/1.1')) {
-    return httpConnectionListener.call(this, socket);
+  if (socket.alpnProtocol === false || socket.alpnProtocol === 'http/1.1') {
+    if (options.allowHTTP1 === true) {
+      // Fallback to HTTP/1.1
+      return httpConnectionListener.call(this, socket);
+    } else {
+      // Reject non-HTTP/2 client
+      return socket.destroy();
```

Comment:
`socket.destroy()`  is actually nice. I'm not sure if we shouldn't even add an `Error` there as first arg.
@jasnell what do you think?